### PR TITLE
[7] 2022/responsive fixes

### PR DESCRIPTION
**Pull request by zastrow**
Addresses many responsive layout and sizing issues from smallest (appx 320px or 20rem) on up to the largest breakpoint (80rem). The primary concer is that content is legible. Please review by resizing the browser with the inspector open in order to get the width down to 320ish pixels, the slowly expand the browser width up, around 400pixels or more you can close the inspector and continue from that size up to at least 1280 or above.

**PLEASE Do Not Use Responsive Mode!**
Responsive mode is neat feature of browsers to mimic different devices sizes but there are two fundamental problems that make the bad for testing:
1. Responsive mode does not show the in between screen sizes that make RWD so beneficial.
2. They often require applying an artificial zoom to the page to fit the emulated device, distorting the page and hiding issues.

If you wish to test this on device, please use [BrowserStack](https://www.browserstack.com/) as an alternative if you do not have any devices on which to test.

### CODE REVIEW

File: src/components/2022/grid-cell.js [hunk 1]
```diff
@@ -10,28 +10,36 @@ const GridCell = (props) => {
   cssProperties['--row-span'] = (props.rowSpan) ? props.rowSpan : null
   cssProperties['--row-start'] = (props.rowStart) ? props.rowStart : null
   cssProperties['--order'] = (props.order) ? props.order : null
```
Comment (lahernandezb) [inline src/components/2022/grid-cell.js:12]: Very minor but all of these could be `cssProperties['--order'] = props.order ?? null` and be shortened up a little.

File: src/components/2022/toc-link.js [hunk 2]
```diff
@@ -2,12 +2,27 @@ import React from "react"
 import PropTypes from "prop-types"
 import AnchorLink from "react-anchor-link-smooth-scroll"
 
+const DownArrow = () => {
+  return (
+    <svg className="cmp-toc-link__arrow" width="24" height="22" fill="none" stroke="currentColor" stroke-width="1.5" xmlns="http://www.w3.org/2000/svg"><path d="M14.768 19.42V1.656H0"/><path d="m22.57 12.213-7.802 7.8-7.8-7.8" /></svg>
```
Comment (mdt2) [inline src/components/2022/toc-link.js:7]: Getting a console error, updating to `strokeWidth` should fix

File: src/scss/2022/components/_hero.scss [hunk 3]
```diff
@@ -24,7 +25,11 @@
     font-family: var(--font-display);
     grid-area: b;
     // Allows the font to be fluid in relation to the hero size
-    font-size: min(14vw, calc(var(--hero-max-width) * 0.14));
+    font-size: min(12vw, calc(var(--hero-max-width) * 0.12));
```
Comment (mdt2) [inline src/scss/2022/components/_hero.scss:28]: Where's 0.12 coming from? (same question for all these)

File: src/scss/2022/components/_hero.scss [hunk 4]
```diff
@@ -49,25 +54,32 @@
...
     // Allows the font to be fluid in relation to the hero size
-    font-size: min(25vw, calc(var(--hero-max-width) * 0.025));
+    font-size: min(5vw, calc(var(--hero-max-width) * 0.025));
     grid-area: c;
     align-self: center;
     color: var(--color-gray);
   }
   
   &__logo {
+    width: 40vw;
```
Comment (mdt2) [inline src/scss/2022/components/_hero.scss:81]: I'm good with this and the max-width values not having a var name as long as it's just used in one place!

File: src/scss/2022/components/_key-finding.scss [hunk 5]
```diff
@@ -16,7 +16,7 @@
   }
   
   &::before {
-    inset: #{space('lg') * -1};
+    inset: -1.5rem;
```
Comment (mdt2) [inline src/scss/2022/components/_key-finding.scss:19]: This is a cool effect 😍

File: src/scss/2022/components/_hero.scss [hunk 6]
```diff
@@ -1,5 +1,6 @@
 @use "../tools/svg" as *;
 @use "../tools/color" as *;
+@use "../tools/media-queries" as *;
 
 .cmp-hero {
```
Comment (lahernandezb) [inline src/scss/2022/components/_hero.scss:5]: Around 476px, the `.cmp-hero__intro` doesn't seem to respect the width of its container `.cmp-hero` and as a result the text bleeds to the edge of the window. 

Screen Shot 2022-06-14 at 10 14 05 AM

**Review by mdt2 — COMMENTED** [review]
It's looking great Philip! Left a couple of comments for you. The only thing I noticed in the browser is a nit-picky/design eye thing. So just a heads up, between 600px-799px these pie charts look really big, especially compared to the others on the page. I recognize that with the layout of the text 800px may be the earliest we can adjust them but just wanted to make you aware:

Screen Shot 2022-06-14 at 10 15 55 AM

**Review by lahernandezb — COMMENTED** [review]
Hey Z, overall everything looks great. There was just one thing I noticed with the hero text.

File: src/components/2022/grid-cell.js (hunk 1 above)
Comment (zastrow) [reply src/components/2022/grid-cell.js:12]: I did not know that! Thank you!

File: src/scss/2022/components/_hero.scss (hunk 3 above)
Comment (zastrow) [reply src/scss/2022/components/_hero.scss:28]: Percentages! So 12vw is 12% of the width of the screen, so the `0.12` is 12% of `var(--hero-max-width)`. It keeps the font-size from growing past 12% of the hero’s `max-width`

File: src/components/2022/toc-link.js (hunk 2 above)
Comment (zastrow) [reply src/components/2022/toc-link.js:7]: Why is React so inconsistent on attributes with dashes?!?! 😩

File: src/scss/2022/components/_hero.scss (hunk 3 above)
Comment (mdt2) [reply src/scss/2022/components/_hero.scss:28]: Ohh of course!! Nice. Thank you!

**Review by lahernandezb — APPROVED** [review]
LGTM!

**Review by mdt2 — APPROVED** [review]
9aw7nnUmkFggE

**Pull request merged**